spec/higher_level_api/integration/connection_recovery_spec.rb

```diff
@@ -40,6 +49,19 @@
     end
   end
 
+  it "calls the network recovery callback" do
+    with_open(c_with_callback) do |c|
+      ch1 = c.create_channel
+      ch2 = c.create_channel
+      sleep 1.5
+      close_all_connections!
+      sleep 0.5
+      poll_until { channels.count == 2 }
+    end
+
+    expect(network_recovery_callback).to have_received(:call).exactly(1).times
```

Comment:
I have strong feelings about using mocks in data service clients such as Bunny (TL;DR: they are overused and mostly useless) but here it might be OK. I'm happy to update this test on my own.

Reply:
here I just passed a double/mock/spy as a callback so i can see if it has been called, how would you approach this otherwise